src/utils/Api.js

```diff
@@ -0,0 +1,104 @@
+class Api {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    getUserInfo() {
+      return fetch(`${this._baseUrl}/users/me`, {
+        headers: { authorization: this._headers.authorization }
+      })
+        .then(res => {
+          if (res.ok) return res.json();
+          return Promise.reject(new Error(`Error: ${res.status}`));
+        })
+        .catch(err => this.handleError(err));
+    }
+  
+    getInitialCards() {
+      return fetch(`${this._baseUrl}/cards`, {
+        headers: { ...this._headers, authorization: this._headers.authorization }
+      })
+        .then(res => {
+          if (res.ok) return res.json();
+          return Promise.reject(new Error(`Error: ${res.status}`));
+        })
+        .catch(err => this.handleError(err));
+    }
+
+    editProfileAvatar({ avatar }) {
+      return fetch(`${this._baseUrl}/users/me/avatar`, {
+        method: "PATCH",
+        headers: { ...this._headers, authorization: this._headers.authorization, "Content-Type": "application/json" },
```

Comment:
`"Content-Type": "application/json"`

According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)